Sources/do-not-disturb/DoNotDisturb.swift

```diff
@@ -12,22 +12,35 @@ public struct DoNotDisturb {
 		DistributedNotificationCenter.default().postNotificationName(NSNotification.Name("com.apple.notificationcenterui.dndprefs_changed"), object: nil, deliverImmediately: true)
 	}
 
-	private static func enable() {
-		set("dndStart", value: 0 as CFPropertyList)
-		set("dndEnd", value: 1440 as CFPropertyList)
-		set("doNotDisturb", value: true as CFPropertyList)
-		commitChanges()
+	private static func restartNotificationCenter() {
+		NSRunningApplication.runningApplications(withBundleIdentifier: "com.apple.notificationcenterui").first?.forceTerminate()
+	}
 
-		// For some reason `doNotDisturb` does not take the first time around
-		// TODO: Figure out why
-		sleep(for: 0.4)
-		set("doNotDisturb", value: true as CFPropertyList)
+	private static func enable() {
+		if !isEnabled {
+			set("doNotDisturb", value: true as CFPropertyList)
+			set("doNotDisturbDate", value: Date() as CFPropertyList)
+			commitChanges()
+			restartNotificationCenter()
+		}
 	}
 
 	private static func disable() {
+		if isEnabled {
+			set("doNotDisturb", value: false as CFPropertyList)
+			set("doNotDisturbDate", value: nil)
+			commitChanges()
+			restartNotificationCenter()
+			restoreMenubarIcon()
+		}
+	}
+
+	private static func restoreMenubarIcon() {
+		set("dndStart", value: 0 as CFPropertyList)
+		set("dndEnd", value: 1440 as CFPropertyList)
+		sleep(for: 0.3)
```

Comment:
Let's make it `0.4` just to be sure. Some systems might be slower. Also needs a comment that `0.3` was the lowest you managed to make it work with and that we're using `0.4` just to be sure.

Seems like the same problem as https://github.com/sindresorhus/macos-wallpaper/blob/cd8b1a7de4432789b58119b08a93cd2a432b77c6/Sources/wallpaper/Wallpaper.swift#L79-L81

Reply:
b69ccf4